provider/machinepool/machine_pool_resource.go

```diff
@@ -125,112 +125,108 @@ func (r *MachinePoolResource) Schema(ctx context.Context, req resource.SchemaReq
 				Description: "The maximum number of replicas for autoscaling functionality.",
 				Optional:    true,
 			},
-			"taints": {
+			"taints": schema.ListNestedAttribute{
 				Description: "Taints for machine pool. Format should be a comma-separated " +
 					"list of 'key=value:ScheduleType'. This list will overwrite any modifications " +
 					"made to node taints on an ongoing basis.\n",
-				Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
-					"key": {
-						Description: "Taints key",
-						Type:        types.StringType,
-						Required:    true,
-					},
-					"value": {
-						Description: "Taints value",
-						Type:        types.StringType,
-						Required:    true,
-					},
-					"schedule_type": {
-						Description: "Taints schedule type",
-						Type:        types.StringType,
-						Required:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Description: "Taints key",
+							Required:    true,
+						},
+						"value": schema.StringAttribute{
+							Description: "Taints value",
+							Required:    true,
+						},
+						"schedule_type": schema.StringAttribute{
+							Description: "Taints schedule type",
+							Required:    true,
+							Validators: []validator.String{
+								stringvalidator.OneOf("NoSchedule", "PreferNoSchedule", "NoExecute"),
+							},
+						},
 					},
-				}, tfsdk.ListNestedAttributesOptions{},
-				),
+				},
 				Optional: true,
 			},
-			"labels": {
+			"labels": schema.MapAttribute{
 				Description: "Labels for the machine pool. Format should be a comma-separated list of 'key = value'." +
 					" This list will overwrite any modifications made to node labels on an ongoing basis.",
-				Type: types.MapType{
-					ElemType: types.StringType,
-				},
-				Optional: true,
+				ElementType: types.StringType,
+				Optional:    true,
 			},
-			"multi_availability_zone": {
+			"multi_availability_zone": schema.BoolAttribute{
 				Description: "Create a multi-AZ machine pool for a multi-AZ cluster (default true)",
-				Type:        types.BoolType,
 				Optional:    true,
 				Computed:    true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					tfsdk.UseStateForUnknown(),
-					ValueCannotBeChangedModifier(),
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+					//ValueCannotBeChangedModifier(),
 				},
 			},
-			"availability_zone": {
+			"availability_zone": schema.StringAttribute{
 				Description: "Select availability zone to create a single AZ machine pool for a multi-AZ cluster",
-				Type:        types.StringType,
 				Optional:    true,
 				Computed:    true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					tfsdk.UseStateForUnknown(),
-					ValueCannotBeChangedModifier(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					//ValueCannotBeChangedModifier(),
```

Comment:
Consider using "RequiresReplace" modifier, like done in first commit

Reply:
done. (I didn't notice :disappointed: )